accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+    "rollForward": "feature",
+  }
+}
+```
+
+This change will only affect `global.json` files with a `version` of `7.0.0` or greater. It will not affect `global.json` files with `6.0.0` (for example) even if .NET 7 SDK is installed on the machine.
+
+The complete set of [roll-forward options are defined in the `global.json` schema](https://docs.microsoft.com/dotnet/core/tools/global-json#rollforward).
+
+### SDK roll-forward gestures
+
+> Customer value: This feature satisfies the "intuitive version selection" and "Convenient and consistent infra builds"customer scenarios.
+
+We will expand the SDK roll-forward gestures to match the similar capabilities for the runtime. These new gestures would enable controlling SDK roll-forward beyond just `global.json`.
+
+It's useful to consider the options that the host offers for runtime roll-forward, via the following ENVs:
+
+- `DOTNET_ROLL_FORWARD` -- a roll-forward policy like `LatestMajor`.
+- `DOTNET_ROLL_FORWARD_TO_PRERELEASE` -- a boolean value on whether roll-forward candidates should include pre-release versions.
+
+The SDK has similar concepts (per an [earlier design](https://github.com/dotnet/designs/blob/main/accepted/2020/global-json-updates.md)). The most obvious approach is to expose matching ENVs as `DOTNET_SDK` to differentiate:
+
+- `DOTNET_SDK_ROLL_FORWARD`
+- `DOTNET_SDK_ROLL_FORWARD_TO_PRERELEASE`
+
+That takes care of the roll-forward ENVs. Let's look at what the CLI offers for the runtime:
+
+- `--roll-forward` -- specifies roll-forward policy
+
+We can follow the same pattern again, by offering a mirror SDK variant:
+
+-- `--sdk-roll-forward`
+
+It is reasonable to ask the question of whether we could simply use the existing runtime-oriented ENVs and CLI switches to also affect the SDK. There are multiple scenarios where users would want separate handling (like with `dotnet test` or `dotnet run`):
+
+- Roll-forward the SDK due to `global.json` pinning but retain runtime targeting as-is (as tested and expected).
+- Leave the SDK as-is (assume it works), but roll-forward the runtime from an end-of-life .NET version to a supported one.
+
+It is also reasonable to ask why we need `--roll-forward` switches for the CLI at all (for the SDK or the runtime). It is useful to specify a minimum version and roll forward policy (via the CLI or otherwise) when you want to enable your code to run in environments you don't control. This could be equally as true for scripts as apps.
+
+Note: The CLI doesn't offer a CLI affordance that matches `DOTNET_ROLL_FORWARD_TO_PRERELEASE`. We should wait for more feedback before adding pre-release-oriented switches to the CLI, for the runtime or SDK.
+
+### Host SDK selection
+
+The host offers `--fx-version` to select a specific runtime. We will add support to the same thing for selecting an SDK with `--sdk-version`.
```

Comment:
There's currently a hidden CLI option for the `dotnet workload` commands called `--sdk-version` - we should verify that there's no unintended interaction between the two. It _should_ be unambiguous if the `--sdk-version` option proposed here can only come between the `dotnet` host and any subcommands/dll arguments/etc.

Reply:
We'll ensure we discuss that with you when it comes time to ensure we make the right decisons.